modules/core/src/v2/wallet.ts

```diff
@@ -2030,12 +2034,12 @@ export class Wallet {
       params.reqId = reqId;
       const coin = self.baseCoin;
       if (_.isObject(params.recipients)) {
-        params.recipients.map(function(recipient) {
+        params.recipients.map(function (recipient) {
           const amount = new BigNumber(recipient.amount);
           if (amount.isNegative()) {
             throw new Error('invalid argument for amount - positive number greater than zero or numeric string expected');
           }
-          if (!coin.valuelessTransferAllowed() && amount.isZero()) {
+          if (!coin.valuelessTransferAllowed() && amount.isZero() && this.typeKeys.includes(params.type)) {
```

Comment:
A better way to implement this could be to pass in an option `type` param in valuelessTransferAllowed and implement it in `algo` so that it checks the param and return true or false.
With this, the logic will be encapsulated in coin implementations.